imagersite/imagersite/tests/test_views.py

```diff
@@ -0,0 +1,50 @@
+from django.contrib.auth.models import User, AnonymousUser
+from django.test import TestCase, RequestFactory
+from imager_images.models import Photo
+from ..views import home_view
+from django.urls import reverse_lazy
+from model_mommy import mommy
+import tempfile
+
+
+class HomeViewTest(TestCase):
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(
+                username="strider",
+                email="strider@aragorn.com",
+                password="kingofgondor",)
+
+    def test_anon_gets_homepage(self):
+        """Test anonymous user lands on home."""
+        request = self.factory.get(reverse_lazy('home'))
+        request.user = AnonymousUser()
+        response = home_view(request)
+
+        self.assertEqual(response.status_code, 200)
```

Comment:
All of the tests where you directly call the view function are unit tests, not functional tests. Functional tests are run through the entire app using the `client`.